src/solana/index.ts

```diff
@@ -24,25 +26,59 @@ export default class SolanaWallet implements IWallet {
   solanaPublicKey: string;
   network: string;
   coin: SolanaCoinData;
+  index: number;
+  accountType: string;
 
-  constructor(xpub: string, coin: SolanaCoinData) {
+  constructor(
+    index: number,
+    accountType: string,
+    xpub: string,
+    coin: SolanaCoinData
+  ) {
+    this.index = index;
+    this.accountType = accountType;
     this.xpub = xpub;
     this.coin = coin;
     this.network = coin.network;
     this.address = generateSolanaAddress(this.xpub);
     this.solanaPublicKey = base_decode(this.address).toString('hex');
   }
 
+  public static getDerivationPath(accountIndex: number, _accountType: string) {
+    let derivationPath =
+      intToUintByte(0x80000000 + 44, 8 * 4) +
+      intToUintByte(0x80000000 + 501, 8 * 4);
+    let derivationDepth = 2;
```

Comment:
For Solana at depth 2 let's check what ledger or at least add conditions in the database to not add for the same derivation path